File: leetcode-04/solutions/cpp/0228.cpp
class Solution {
 public:
  vector<string> summaryRanges(vector<int>& nums) {
    const int n = nums.size();

    vector<string> ans;

    for (int i = 0; i < n; i++) {
      int start = nums[i];
      while (i < n - 1 && nums[i] == nums[i + 1] - 1) i++;
      int end = nums[i];
      if (start == end)
        ans.push_back(to_string(start));
      else
        ans.push_back(to_string(start) + "->" + to_string(end));
    }

    return ans;
  }
};

File: leetcode-04/solutions/python3/0228.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        n = len(nums)

        ans = []
        i = 0

        while i < n:
            start = nums[i]
            while i < n - 1 and nums[i] == nums[i + 1] - 1:
                i += 1
            end = nums[i]
            if start == end:
                ans.append(str(start))
            else:
                ans.append(str(start) + "->" + str(end))
            i += 1

        return ans
